test-suite/src/test/scala/org/scalajs/testsuite/javalib/ConcurrentHashMapTest.scala

```diff
@@ -0,0 +1,110 @@
+package org.scalajs.testsuite.javalib
+
+import language.implicitConversions
+
+import scala.language.implicitConversions
+
+import org.scalajs.jasminetest.JasmineTest
+
+object ConcurrentHashMapTest extends JasmineTest {
+
+  describe("java.util.concurrent.ConcurrentHashMap") {
+
+    it("should store strings") {
+      val chm = new java.util.concurrent.ConcurrentHashMap[String, String]()
+      assert ( chm != null )
+
+      expect(chm.size()).toEqual(0)
+      chm.put("ONE", "one")
+      expect(chm.size()).toEqual(1)
+      expect(chm.get("ONE")).toEqual("one")
+      chm.put("TWO", "two")
+      expect(chm.size()).toEqual(2)
+      expect(chm.get("TWO")).toEqual("two")
+    }
+
+    it("should store integers") {
+      val chm = new java.util.concurrent.ConcurrentHashMap[Int, Int]()
+      assert ( chm != null )
+
+      chm.put(1, 1)
+      expect(chm.size()).toEqual(1)
+      val one = chm.get(1)
+      expect(one).toEqual(1)
+    }
+
+    it("should store custom objects") {
+      case class TestObj(num: Int)
+
+      val chm = new java.util.concurrent.ConcurrentHashMap[TestObj, TestObj]()
+      assert ( chm != null )
+
+      chm.put(TestObj(1), TestObj(1))
+      expect(chm.size()).toEqual(1)
+      val one = chm.get(TestObj(1))
+      expect(one.num).toEqual(1)
+    }
+
+    it("should remove stored elements") {
+      val chm = new java.util.concurrent.ConcurrentHashMap[String, String]()
+      assert ( chm != null )
+
+      chm.put("ONE", "one")
+      expect(chm.size()).toEqual(1)
+      expect(chm.remove("ONE")).toEqual("one")
+      val newOne = chm.get("ONE")
+      expect(chm.get("ONE")).toBeNull
+    }
+
+    it("should behave like expected on every method") {
```

Comment:
You should separate your tests per method. Each test should test the common and corner cases of the method they test.

In particular pay attention to `null`s, as they typically behave differently in Java collections than in Scala collections (mostly as keys).